packages/sdk/src/utils/contracts.ts

```diff
@@ -26,8 +26,10 @@ class BaseContract extends BaseContractEthers {}
 
 export type BaseContractType<TAbi extends Abi> = BaseContract & ContractMethods<TAbi>
 
+export type AnyFunction<TArgs extends any[], TReturn extends any> = (...params: TArgs) => TReturn
+
 export type HolographByNetworksResponse = {
-  [chainId: number]: string | number | BigInt //TODO: Change to `string` as soon as we fix FunctionReturnTypes in the file '../utils/contracts.ts'
+  [chainId: number]: string | string[] //TODO: work on it
```

Comment:
Can you provide the updated details of the issue so we can remember what we need to work on / figure out for this?

Reply:
There're two things that need to be deal with:

1. We need to better define the return type for our functions. It's my understanding that we want to create our own type instead of the type returned when making the raw call.

2. The raw call return types are not correctly mapped using ABITypes, we need to make some improvements in our type definitions.